server_src/lib/dataFunctions/printerClean.js

```diff
@@ -416,7 +416,7 @@ class PrinterClean {
       );
       sortedPrinter.connectionOptions = farmPrinter.options;
       if (
-        !!sortedPrinter?.connectionOptions &&
+        !!sortedPrinter?.connectionOptions?.ports &&
```

Comment:
Bugfix. For my VIRTUAL only OP instance there was no ports property. Could be that this array is not sent when only 1 connection exists? OP version 1.6+